File: postgres_39/struct/SeisBinGrid.go
package dto

import (
	"time"
)

type Seis_bin_grid struct {
	Seis_set_subtype       string     `json:"seis_set_subtype"`
	Seis_set_id            string     `json:"seis_set_id"`
	Bin_grid_id            string     `json:"bin_grid_id"`
	Source                 string     `json:"source"`
	Active_ind             *string    `json:"active_ind"`
	Angle_rotation         *float64   `json:"angle_rotation"`
	Bin_grid_type          *string    `json:"bin_grid_type"`
	Bin_grid_version       *float64   `json:"bin_grid_version"`
	Bin_method             *string    `json:"bin_method"`
	Bin_point_ouom         *string    `json:"bin_point_ouom"`
	Coord_acquisition_id   *string    `json:"coord_acquisition_id"`
	Coord_system_id        *string    `json:"coord_system_id"`
	Corner1_lat            *float64   `json:"corner_1_lat"`
	Corner1_long           *float64   `json:"corner_1_long"`
	Corner2_lat            *float64   `json:"corner_2_lat"`
	Corner2_long           *float64   `json:"corner_2_long"`
	Corner3_lat            *float64   `json:"corner_3_lat"`
	Corner3_long           *float64   `json:"corner_3_long"`
	Effective_date         *time.Time `json:"effective_date"`
	Expiry_date            *time.Time `json:"expiry_date"`
	Fold_coverage          *float64   `json:"fold_coverage"`
	Local_coord_system_id  *string    `json:"local_coord_system_id"`
	Nline_azimuth          *float64   `json:"nline_azimuth"`
	Nline_count            *int       `json:"nline_count"`
	Nline_max_no           *int       `json:"nline_max_no"`
	Nline_min_no           *int       `json:"nline_min_no"`
	Nline_spacing          *float64   `json:"nline_spacing"`
	North_type             *string    `json:"north_type"`
	Point_origin_easting   *float64   `json:"point_origin_easting"`
	Point_origin_latitude  *float64   `json:"point_origin_latitude"`
	Point_origin_longitude *float64   `json:"point_origin_longitude"`
	Point_origin_northing  *float64   `json:"point_origin_northing"`
	Ppdm_guid              string     `json:"ppdm_guid"`
	Remark                 *string    `json:"remark"`
	Xline_azimuth          *float64   `json:"xline_azimuth"`
	Xline_count            *int       `json:"xline_count"`
	Xline_max_no           *int       `json:"xline_max_no"`
	Xline_min_no           *int       `json:"xline_min_no"`
	Xline_spacing          *float64   `json:"xline_spacing"`
	Row_changed_by         *string    `json:"row_changed_by"`
	Row_changed_date       *time.Time `json:"row_changed_date"`
	Row_created_by         *string    `json:"row_created_by"`
	Row_created_date       *time.Time `json:"row_created_date"`
	Row_effective_date     *time.Time `json:"row_effective_date"`
	Row_expiry_date        *time.Time `json:"row_expiry_date"`
	Row_quality            *string    `json:"row_quality"`
}

File: create_tables/seis_bin_grid.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table seis_bin_grid';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """create table seis_bin_grid(
        SEIS_SET_SUBTYPE VARCHAR(30) NOT NULL,
        SEIS_SET_ID VARCHAR(40) NOT NULL,
        BIN_GRID_ID VARCHAR(40) NOT NULL,
        SOURCE VARCHAR(40) NOT NULL,
        ACTIVE_IND VARCHAR(1),
        ANGLE_ROTATION DECIMAL(10,5),
        BIN_GRID_TYPE VARCHAR(40),
        BIN_GRID_VERSION DECIMAL(5,2),
        BIN_METHOD VARCHAR(40),
        BIN_POINT_OUOM VARCHAR(40),
        COORD_ACQUISITION_ID VARCHAR(40),
        COORD_SYSTEM_ID VARCHAR(40),
        CORNER1_LAT DECIMAL(14,9),
        CORNER1_LONG DECIMAL(14,9),
        CORNER2_LAT DECIMAL(14,9),
        CORNER2_LONG DECIMAL(14,9),
        CORNER3_LAT DECIMAL(14,9),
        CORNER3_LONG DECIMAL(14,9),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        FOLD_COVERAGE DECIMAL(10,5),
        LOCAL_COORD_SYSTEM_ID VARCHAR(40),
        NLINE_AZIMUTH DECIMAL(10,5),
        NLINE_COUNT INT,
        NLINE_MAX_NO INT,
        NLINE_MIN_NO INT,
        NLINE_SPACING DECIMAL(10,3),
        NORTH_TYPE VARCHAR(40),
        POINT_ORIGIN_EASTING DECIMAL(15,7),
        POINT_ORIGIN_LATITUDE DECIMAL(14,9),
        POINT_ORIGIN_LONGITUDE DECIMAL(14,9),
        POINT_ORIGIN_NORTHING DECIMAL(15,7),
        PPDM_GUID VARCHAR(38),
        REMARK VARCHAR(2000),
        XLINE_AZIMUTH DECIMAL(10,5),
        XLINE_COUNT INT,
        XLINE_MAX_NO INT,
        XLINE_MIN_NO INT,
        XLINE_SPACING DECIMAL(10,3),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")